File: backend/src/main/java/com/tribly/enums/RouteSortBy.java
package com.tribly.enums;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public enum RouteSortBy {
  DISTANCE("te.distance"),
  ELEVATION_GAIN("te.elevationGain"),
  HILLINESS("te.hilliness"),
  DATE_TIME("te.dateTime");

  final String field;
}

File: frontend/src/api/dto/routeSortBy.ts
export type RouteSortBy = (typeof RouteSortBy)[keyof typeof RouteSortBy]

export const RouteSortBy = {
  DISTANCE: 'DISTANCE',
  ELEVATION_GAIN: 'ELEVATION_GAIN',
  HILLINESS: 'HILLINESS',
  DATE_TIME: 'DATE_TIME',
} as const
